test/dart2_9_suggestors/factory_and_config_ignore_comment_remover_test.dart

```diff
@@ -0,0 +1,553 @@
+          ''',
+          expectedOutput: '''
+            UiFactory<FooProps> Foo = _\$Foo;
+          ''',
+        );
+      });
+
+      test('when there is another comment above it', () {
+        testSuggestor(
+          expectedPatchCount: 1,
+          input: '''
+            // this is another comment
+            // ignore: $ignoreToRemove
+            UiFactory<FooProps> Foo = _\$Foo; 
+          ''',
+          expectedOutput: '''
+            // this is another comment
+            UiFactory<FooProps> Foo = _\$Foo; 
+          ''',
+        );
+      });
+
+      test('when there is a doc comment above it', () {
+        testSuggestor(
+          expectedPatchCount: 1,
+          input: '''
+            /// this is a doc comment
+            // ignore: $ignoreToRemove
+            UiFactory<FooProps> Foo = _\$Foo; 
+          ''',
+          expectedOutput: '''
+            /// this is a doc comment
+            UiFactory<FooProps> Foo = _\$Foo; 
+          ''',
+        );
+      });
+
+      test('when there are multiple factories', () {
+        testSuggestor(
+          expectedPatchCount: 3,
+          input: '''
+            UiFactory<FooProps> Foo = _\$Foo; // ignore: $ignoreToRemove
+            
+            UiFactory<BarProps> Bar = 
+              // ignore: $ignoreToRemove
+              _\$Bar; 
+            
+            // ignore: $ignoreToRemove
+            UiFactory<BazProps> Baz = _\$Baz;
+          ''',
+          expectedOutput: '''
+            UiFactory<FooProps> Foo = _\$Foo;
+            
+            UiFactory<BarProps> Bar = _\$Bar;
+            
+            UiFactory<BazProps> Baz = _\$Baz;
+          ''',
+        );
+      });
+
+      group('when there are multiple ignores', () {
+        test('in the same comment', () {
+          testSuggestor(
+            expectedPatchCount: 1,
+            input: '''
+              UiFactory<FooProps> Foo = _\$Foo; // ignore: invalid_assignment, $ignoreToRemove, unused_element
+            ''',
+            expectedOutput: '''
+              UiFactory<FooProps> Foo = _\$Foo; // ignore: invalid_assignment, unused_element
+            ''',
+          );
+        });
+
+        test('in different comments', () {
+          testSuggestor(
+            expectedPatchCount: 3,
+            input: '''
+              // ignore: $ignoreToRemove, unused_element
+              UiFactory<FooProps> Foo = 
+                // ignore: $ignoreToRemove
+                _\$Foo; // ignore: invalid_assignment, $ignoreToRemove
+            ''',
+            expectedOutput: '''
+              // ignore: unused_element
+              UiFactory<FooProps> Foo = _\$Foo; // ignore: invalid_assignment
+            ''',
+          );
+        });
+      });
+
+      test('when the factory is already updated', () {
+        testSuggestor(
+          expectedPatchCount: 1,
+          input: '''
+            UiFactory<FooProps> Foo = $castFunctionName(_\$Foo); // ignore: $ignoreToRemove
```

Comment:
Shouldn't the codemod keep `undefined_identifier` ignores in place for Dart <2.12 compatibility?

Reply:
Oh wait, is it because you're calling `ignoreRemoverTestHelper('undefined_identifier');` in the test, but not passing in `'undefined_identifier'` in the real codemod?

Reply:
Yep! I just tested the suggestor using `undefined_identifier` because that was the case where the ignore comments would be in all those locations in real life, but I also tested `ignoreRemoverTestHelper('argument_type_not_assignable')` to verify that it removes whichever comment is given in the constructor. But, yes, the real codemod is [only removing](https://github.com/Workiva/over_react_codemod/blob/17b25b1ee71fc4e27fed47d9ebca3e593ef36e64/lib/src/executables/dart2_9_upgrade.dart#L55-L56) `invalid_assignment` and `argument_type_not_assignable` right now.